tests/phpstan-baseline.neon

```diff
@@ -0,0 +1,49 @@
+parameters:
+	ignoreErrors:
+		-
+			message: '#^Cannot call method matches\(\) on Hamcrest\\Matcher\|null\.$#'
+			identifier: method.nonObject
+			count: 1
+			path: ../hamcrest/Hamcrest/Arrays/SeriesMatchingOnce.php
+
+		-
+			message: '#^Parameter \#1 \$matcher of method Hamcrest\\Arrays\\SeriesMatchingOnce\:\:_describeMismatch\(\) expects Hamcrest\\Matcher, Hamcrest\\Matcher\|null given\.$#'
+			identifier: argument.type
+			count: 1
+			path: ../hamcrest/Hamcrest/Arrays/SeriesMatchingOnce.php
+
+		-
+			message: '#^Cannot call method toString\(\) on class\-string\|object\.$#'
+			identifier: method.nonObject
+			count: 1
+			path: ../hamcrest/Hamcrest/Core/HasToString.php
+
+		-
+			message: '#^Cannot cast Hamcrest\\BaseDescription to string\.$#'
+			identifier: cast.string
+			count: 1
+			path: ../hamcrest/Hamcrest/StringDescription.php
```

Comment:
i think this does only work because SelfDescribing is mostly used with the BaseMatcher and the basematcher implements __toString()
Imho this needs to add __toString() to the SelfDescribing interface

Reply:
Please check this in Java version of the library ( https://github.com/hamcrest/JavaHamcrest ) from where this code was ported.

To make sure it continues to work please also add test to uncover this bad behavior. This way we can be sure, that it won't come back in the future.

Reply:
Depending on what you want to do with this monster-PR I was planning to get to this issues one by one so that they even can be backported to the 2.x branch.

Reply:
You're right (that `static` should used as the `\Hamcrest\BaseDescription::appendDescriptionOf` return type instead of the `self` to fix the issue. Since `static` return type only works since PHP 8.0, then we'll need to use it via a DocBlocks.

Reply:
sure! done